tests/unit/test_column_noise.py

```diff
@@ -156,9 +163,55 @@ def test_swap_months_and_days():
     pass
 
 
-@pytest.mark.skip(reason="TODO")
-def test_miswrite_zipcodes():
-    pass
+def test_miswrite_zipcodes(dummy_dataset):
+    config = get_configuration()
+    config.update(
+        {
+            "decennial_census": {
+                "zipcode": {
+                    "zipcode_miswriting": {
+                        "row_noise_level": 0.5,
+                        "first_two_digits_noise_leveL": 0.3,
+                        "middle_digit_noise_leveL": 0.4,
+                        "last_two_digits_noise_leveL": 0.5,
+                    },
+                },
+            },
+        }
+    )
+    config = config["decennial_census"]["zipcode"]["zipcode_miswriting"]
+
+    # Get configuration values for each piece of 5 digit zipcode
+    row_noise_level = config.row_noise_level
+    first2_prob = config.first_two_digits_noise_leveL
+    middle_prob = config.middle_digit_noise_leveL
+    last2_prob = config.last_two_digits_noise_leveL
+    data = dummy_dataset["zipcode"]
+    noised_data = NOISE_TYPES.ZIPCODE_MISWRITING(data, config, RANDOMNESS0, "test_zipcode")
+
+    # Confirm missing data remains missing
+    orig_missing = data == ""
+    assert (noised_data[orig_missing] == "").all()
+    # Check noise for each digits position matches expected noise
+    for i in range(5):  # "12345"
```

Comment:
Nit: since you have three different cases for the five values, you're not getting much value out of this loop. This could be cleaner:
```
for i in range(2):
    assert np.isclose(
        first_two_digits_condition
    )

assert np.isclose(
    middle_prob * row_noise_level,
    (data[~orig_missing].str[2] != noised_data[~orig_missing].str[2]).mean(),
    rtol=0.02,
)


for i in range(3, 5):
    assert np.isclose(
        last_two_digits_condition
    )
```